pytype/tools/traces/traces_test.py

```diff
@@ -283,28 +283,40 @@ def f(x, y, z):
 class MatchConstantTest(MatchAstTestCase):
 
   def test_num(self):
-    matches = self._get_traces("v = 42", ast.Num)
+    matches = self._get_traces(
+        "v = 42",
+        ast.Constant if sys.version_info >= (3, 8) else ast.Num)
     self.assertTracesEqual(matches, [((1, 4), "LOAD_CONST", 42, ("int",))])
 
   def test_str(self):
-    matches = self._get_traces("v = 'hello'", ast.Str)
+    matches = self._get_traces(
+        "v = 'hello'",
+        ast.Constant if sys.version_info >= (3, 8) else ast.Str)
```

Comment:
break all these into two tests, e.g. `test_str` and `test_str_38`, and guard the tests with `@testutils.skipBeforePy`/`@testutils.skipAfterPy` (see `test_literal_37` above for an example).